lib/jekyll/filters.rb

```diff
@@ -345,16 +347,9 @@ def sort_input(input, property, order)
 
     private
     def time(input)
-      case input
-      when Time
-        input.clone
-      when Date
-        input.to_time
-      when String
-        Time.parse(input) rescue Time.at(input.to_i)
-      when Numeric
-        Time.at(input)
-      else
+      begin
+        Liquid::Utils.to_date(input).to_time
+      rescue NoMethodError
```

Comment:
I'd rather us check a return value of to_date than rescue such a generic error here.